tests/query_runner/test_script.py

```diff
@@ -17,3 +18,19 @@ def test_specified(self):
 
         self.monkeypatch.setattr(os.path, "exists", lambda x: True)
         self.assertEqual(["/foo/bar/baz.sh"], query_to_script_path("/foo/bar", "baz.sh"))
+
+
+class TestRunScript(BaseTestCase):
+    monkeypatch = MonkeyPatch()
+
+    def test_success(self):
+        self.monkeypatch.setattr(subprocess, "check_output", lambda script, shell: "test")
+        self.assertEqual(("test", None), run_script("/foo/bar/baz.sh", True))
+
+    def test_failure(self):
+        self.monkeypatch.setattr(subprocess, "check_output", lambda script, shell: None)
+        self.assertEqual((None, "Error reading output"), run_script("/foo/bar/baz.sh", True))
+        self.monkeypatch.setattr(subprocess, "check_output", lambda script, shell: "")
+        self.assertEqual((None, "Error reading output"), run_script("/foo/bar/baz.sh", True))
```

Comment:
Maybe return "Empty output from script." message in this (and the `" "`) case?

Reply:
Thank you, it makes sense.